akvo/rest/views/user_management.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""Akvo RSR is covered by the GNU Affero General Public License.
+
+See more details in the license.txt file located at the root folder of the Akvo RSR module.
+For additional details on the GNU license please see < http://www.gnu.org/licenses/agpl.html >.
+"""
+
+import json
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import Group
+from django.core.exceptions import ValidationError
+from django.core.signing import TimestampSigner
+from django.core.validators import validate_email
+from django.db import IntegrityError
+
+from rest_framework import status
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from akvo.utils import rsr_send_mail
+from ...rsr.models import Employment, Organisation
+
+
+@api_view(['POST'])
+@permission_classes((IsAuthenticated, ))
+def invite_user(request):
+    """
+    Invite a new user to RSR.
+
+    :param request; request.DATA is a JSON string containing email, organisation and group data
+    """
+    def valid_email(email_address):
+        try:
+            validate_email(email_address)
+            return True
+        except ValidationError:
+            return False
+
+    user = request.user
+    if not user.has_perm('rsr.user_management'):
+        return Response('Request not allowed', status=status.HTTP_403_FORBIDDEN)
+
+    # Check if all information is present, and if the organisation and group exist
+    data, missing_data = request.DATA.get('user_data'), []
+    if not data:
+        return Response({'missing_data': missing_data.append(['email', 'organisation', 'group'])},
+                        status=status.HTTP_400_BAD_REQUEST)
+    else:
+        data = json.loads(data)
+
+    # Check email
+    email = data['email'].lower().strip()
+    if not (email and valid_email(email) and user.email != email):
+        missing_data.append('email')
+
+    # Check organisation
+    try:
+        organisation = Organisation.objects.get(pk=int(data['organisation']))
+    except (Organisation.DoesNotExist, ValueError):
+        missing_data.append('organisation')
+
+    # Check group
+    try:
+        group = Group.objects.get(pk=int(data['group']))
+    except (Group.DoesNotExist, ValueError):
+        missing_data.append('group')
+
+    if missing_data:
+        return Response({'missing_data': missing_data}, status=status.HTTP_400_BAD_REQUEST)
+
+    # Check if the user already exists, based on the email address
+    try:
+        invited_user = get_user_model().objects.get(email=email)
+    except get_user_model().DoesNotExist:
+        try:
+            invited_user = get_user_model().objects.create_user(email, email=email)
+            invited_user.set_is_active(False)
```

Comment:
Yeah, I agree. I've now changed the default setting of the ```create_user()``` function to create inactive users, since it's kind of weird to create active users. I checked and user registration is working as intended and the function isn't used anywhere else.